.ci/scripts/go-plus/github-differ/generate_comment.sh

```diff
@@ -45,73 +45,60 @@ if ! git diff --exit-code origin/$OLD_BRANCH origin/$NEW_BRANCH; then
     SUMMARY=`git diff origin/$OLD_BRANCH origin/$NEW_BRANCH --shortstat`
     DIFFS="${DIFFS}${NEWLINE}Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/$OLD_BRANCH..$NEW_BRANCH) ($SUMMARY)"
 fi
-TPG_LOCAL_PATH_OLD="${TPG_LOCAL_PATH}old"
-mkdir -p $TPG_LOCAL_PATH_OLD
-cp -r $TPG_LOCAL_PATH/. $TPG_LOCAL_PATH_OLD
+popd
 
-git checkout origin/$NEW_BRANCH
-update_package_name "github.com/hashicorp/terraform-provider-google" "google/provider/new"
+mkdir -p $TPGB_LOCAL_PATH
+git clone -b $NEW_BRANCH $TPGB_SCRATCH_PATH $TPGB_LOCAL_PATH
+pushd $TPGB_LOCAL_PATH
+git fetch origin $OLD_BRANCH
+if ! git diff --exit-code origin/$OLD_BRANCH origin/$NEW_BRANCH; then
+    SUMMARY=`git diff origin/$OLD_BRANCH origin/$NEW_BRANCH --shortstat`
+    DIFFS="${DIFFS}${NEWLINE}Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/$OLD_BRANCH..$NEW_BRANCH) ($SUMMARY)"
+fi
 popd
 
 ## Breaking change setup and execution
-pushd $TPG_LOCAL_PATH_OLD
-git checkout origin/$OLD_BRANCH
-update_package_name "github.com/hashicorp/terraform-provider-google" "google/provider/old"
-popd
 set +e
 pushd $MM_LOCAL_PATH/tools/breaking-change-detector
-sed -i.bak -E "s~google/provider/(.*)/([0-9A-Za-z-]*)~google/provider/\1/google~" comparison.go
-go mod edit -replace google/provider/new=$(realpath $TPG_LOCAL_PATH)
-go mod edit -replace google/provider/old=$(realpath $TPG_LOCAL_PATH_OLD)
-go mod tidy
-export TPG_BREAKING="$(go run .)"
+cp -r $TPG_LOCAL_PATH old/
+cp -r $TPG_LOCAL_PATH new/
+make build OLD_REF=$OLD_BRANCH NEW_REF=$NEW_BRANCH
+export TPG_BREAKING="$(bin/breaking-change-detector)"
 retVal=$?
 if [ $retVal -ne 0 ]; then
     export TPG_BREAKING=""
     BREAKING_CHANGE_BUILD_FAILURE=$?
 fi
-set -e
-popd
 
-# TPGB
-mkdir -p $TPGB_LOCAL_PATH
-git clone -b $NEW_BRANCH $TPGB_SCRATCH_PATH $TPGB_LOCAL_PATH
-pushd $TPGB_LOCAL_PATH
-git fetch origin $OLD_BRANCH
-if ! git diff --exit-code origin/$OLD_BRANCH origin/$NEW_BRANCH; then
-    SUMMARY=`git diff origin/$OLD_BRANCH origin/$NEW_BRANCH --shortstat`
-    DIFFS="${DIFFS}${NEWLINE}Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/$OLD_BRANCH..$NEW_BRANCH) ($SUMMARY)"
+rm -rf ./old/ ./new/
+cp -r $TPGB_LOCAL_PATH old/
+cp -r $TPGB_LOCAL_PATH new/
+make build OLD_REF=$OLD_BRANCH NEW_REF=$NEW_BRANCH
+export TPGB_BREAKING="$(bin/breaking-change-detector)"
+retVal=$?
+if [ $retVal -ne 0 ]; then
+    export TPGB_BREAKING=""
+    BREAKING_CHANGE_BUILD_FAILURE=$?
```

Comment:
`BREAKING_CHANGE_BUILD_FAILURE` was meant to catch if the build fails to run for a given customer change. Then we would run the unit tests for the change. Since the units tests are no longer based on the customer PR we can effectively remove this flag. Although, we likely want to say something to customer in case the tool fails to run.

ie : breaking-change-detector crashed durring execution..

We've had instances before where breaking-change-detector would silently break/fail. 

This is why we run unit tests if there was a crash.

Reply:
Ideally breaking change detector would be running in a workflow_dispatch so that any build failures failure / breaking changes would be logged there rather than included in the diff output, which would fix this problem. However, this is already a pretty big PR; do you have a preference for making that change here vs in a separate PR?

Reply:
If we are removing unit tests off the customer change we need to output that the breaking change detector crashed inline somewhere. I can make a suggestion on how.. one second

Reply:
💯 makes sense, thanks!